sktime/forecasting/base/adapters/_statsmodels.py

```diff
@@ -111,12 +111,15 @@ def _predict(self, fh, X):
         # statsmodels exog must contain test indices only.
         # For discussion see https://github.com/sktime/sktime/issues/3830
         if X is not None:
-            ind_drop = self._X.index
-            X = X.loc[~X.index.isin(ind_drop)]
+            if self._X is not None:
+                ind_drop = self._X.index
+                X = X.loc[~X.index.isin(ind_drop)]
             # Entire range of the forecast horizon is required
             X = X.iloc[: (fh_int[-1] + 1)]  # include end point
 
         if "exog" in inspect.signature(self._forecaster.__init__).parameters.keys():
+            if self._X is None:
```

Comment:
I'm not sure if I'm clear about when this check will evaluate to True. Can you please explain?

Reply:
This will evaluate to True when the user has not passed X to fit (hence self._X is None) but has passed X into predict.

Reply:
I was thinking of another estimator, where "exog" will be present in the fitted forecaster depending on whether it's trained with X or not. I checked now with SARIMAX/VAR/VARMAX and that doesn't seemed to the case, and I can't remember where I saw that type of behaviour. Sorry for the confusion.